ui/src/main/kotlin/org/jetbrains/jewel/ui/component/Button.kt

```diff
@@ -102,62 +153,377 @@ public fun OutlinedButton(
+            content = content,
+            secondaryContent = {
+                SplitButtonChevron(
+                    style = style,
+                    enabled = enabled,
+                    isDefault = isDefault,
+                    onChevronClicked = {
+                        secondaryOnClick()
+                        popupVisible = !popupVisible
+                    }
+                )
+            }
+        )
+
+        if (popupVisible && enabled) {
+            if (secondaryContentMenu != null) {
+                PopupMenu(
+                    modifier = Modifier
+                        .width(buttonWidth)
+                        .onClick { popupVisible = false },
+                    onDismissRequest = {
+                        popupVisible = false
+                        true
+                    },
+                    horizontalAlignment = Alignment.Start,
+                    style = menuStyle,
+                    content = secondaryContentMenu,
+                )
+            }
+            if (secondaryContent != null) {
+                PopupContainer(
+                    modifier = Modifier
+                        .width(buttonWidth)
+                        .onClick { popupVisible = false },
+                    onDismissRequest = { popupVisible = false },
+                    horizontalAlignment = Alignment.Start,
+                    content = secondaryContent,
                 )
-                Icon(AllIconsKeys.General.ChevronDown, "Chevron")
             }
         }
-    )
+    }
+}
+
+@Composable
+private fun SplitButtonChevron(
+    style: SplitButtonStyle,
+    enabled: Boolean,
+    isDefault: Boolean,
+    onChevronClicked: () -> Unit,
+) {
+    Box(
+        Modifier
+            .size(style.button.metrics.minSize.height)
+            .focusable(false)
+            .focusProperties { canFocus = false }
+            .clickable(
+                enabled = enabled,
+                onClick = onChevronClicked,
+                interactionSource = remember { MutableInteractionSource() },
+                indication = null
+            )
+    ) {
+        Divider(
+            orientation = Orientation.Vertical,
+            thickness = style.dividerMetrics.thickness,
+            modifier = Modifier
+                .fillMaxHeight()
+                .padding(vertical = style.dividerPadding)
+                .align(Alignment.CenterStart),
+            color = style.dividerColor,
+        )
+        Icon(
+            key = AllIconsKeys.General.ChevronDown,
+            contentDescription = "Chevron",
+            modifier = Modifier.align(Alignment.Center),
+            hints = if (isDefault) {
+                arrayOf(PainterHintStroke(style.chevronColor))
+            } else {
+                emptyArray()
+            }
+        )
+    }
+}
+
+private fun splitButtonKeys(
+    keyEvent: KeyEvent,
+    popupVisible: Boolean,
+    collapsePopup: () -> Unit,
+    expandPopup: () -> Unit,
```

Comment:
Let's align naming and verbs with the rest of the file:

```suggestion
    onSetPopupVisible: (Boolean) -> Unit,
```